token/core/fabtoken/driver/driver.go

```diff
@@ -111,21 +114,26 @@ func (d *Driver) NewTokenService(sp driver.ServiceProvider, networkID string, ch
 		return nil, errors.Wrapf(err, "failed to get identity storage provider")
 	}
 	ip := identity.NewProvider(sigService, view.GetEndpointService(sp), NewEnrollmentIDDeserializer(), deserializerManager)
-	ws := fabtoken.NewWalletService(
-		ip,
+	publicParamsManager := fabtoken.NewPublicParamsManager(
+		fabtoken.PublicParameters,
 		qe,
-		NewDeserializer(),
+	)
+	deserializer := NewDeserializer()
+	ws := common.NewWalletService(
+		logger,
+		ip,
+		func() (driver.Deserializer, error) {
```

Comment:
If you think that this getter might not be used, you could use `NewLazyGetter(NewDeserializer)`

Reply:
if you are okay, I'll change this in a later PR where I'll extract common parts of the token service.